### PR TITLE
CBL-3148: Add c4listener_shareCollection API

### PR DESCRIPTION
**Pull request by borrrden**
:new: API added c4listener_shareCollection (and c++ equivalent)
:new: API added c4listener_unshareCollection (and c++ equivalent)

Note that this API is additive to c4listener_shareDB, and not a replacement.  See the interface document for more details.

### CODE REVIEW

**Comment by borrrden** [comment]
I am going to add tests as another commit, to keep the review focused on the implementation.

File: REST/Listener.cc [hunk 1]
```diff
@@ -98,6 +104,40 @@ namespace litecore { namespace REST {
...
+            _allowedCollections.emplace(name, collections);
+        } else {
+            j->second.push_back(collection);
+        }
+        
+        return true;
+    }
+
+
+    bool Listener::unregisterCollection(const string& name, C4CollectionSpec collection) {
```
Comment (pasin) [inline REST/Listener.cc:125]: Use CollectionSpec for consistent with other C++ code? It also implements == operator.

File: REST/Listener.hh [hunk 2]
```diff
@@ -54,6 +55,13 @@ namespace litecore { namespace REST {
 
         bool unregisterDatabase(C4Database *db);
 
+         /** Adds a collection to be used by Sync Listener on a given database shared via name.  
+             Only collections that are registered will be replicated.  
+             If none are registered, the default collection will be replicated */
+        bool registerCollection(const std::string& name, C4CollectionSpec collection);
```
Comment (pasin) [inline REST/Listener.hh:61]: Use CollectionSpec for consistency with other C++ API.

File: REST/RESTListener+Handlers.cc [hunk 3]
```diff
@@ -76,10 +76,10 @@ namespace litecore { namespace REST {
 #pragma mark - DATABASE HANDLERS:
 
     
-    void RESTListener::handleGetDatabase(RequestResponse &rq, C4Database *db) {
-        auto docCount = db->getDocumentCount();  // TODO: Collection-aware
-        auto lastSequence = db->getLastSequence();
-        C4UUID uuid = db->getPublicUUID();
+    void RESTListener::handleGetDatabase(RequestResponse &rq, C4Collection *coll) {
```
Comment (pasin) [inline REST/RESTListener+Handlers.cc:79]: Does it require to have scope and collection name in the result?

Comment (borrrden) [reply REST/RESTListener+Handlers.cc:79]: I don't think so, since the user is requesting a specific collection via URL

**Review by pasin — APPROVED** [review]

**Pull request merged**